frank/fit.py

```diff
@@ -364,145 +359,164 @@ def output_results(u, v, vis, weights, sol, iteration_diag, iter_plot_range,
-                              model['geometry']['dra'],
-                              model['geometry']['ddec'],
-                              model['geometry']['type'],
-                              model['geometry']['fit_phase_offset']
-                              )
-
-    sol, iteration_diagnostics = perform_fit(u, v, vis, weights, geom,
-                                             model['hyperpriors']['rout'],
-                                             model['hyperpriors']['n'],
-                                             model['hyperpriors']['alpha'],
-                                             model['hyperpriors']['wsmooth'],
-                                             model['hyperpriors']['iter_tol'],
-                                             model['hyperpriors']['max_iter'],
-                                             model['input_output']['iteration_diag'],
-                                             model['plotting']['diag_plot']
-                                             )
-
-    figs, axes = output_results(u, v, vis, weights, sol, iteration_diagnostics,
-                          model['plotting']['iter_plot_range'],
-                          model['plotting']['bin_widths'],
-                          model['input_output']['format'],
-                          model['input_output']['save_prefix'],
-                          model['input_output']['save_profile_fit'],
-                          model['input_output']['save_vis_fit'],
-                          model['input_output']['save_uvtables'],
-                          model['input_output']['iteration_diag'],
-                          model['plotting']['full_plot'],
-                          model['plotting']['quick_plot'],
-                          model['plotting']['diag_plot'],
-                          model['plotting']['force_style'],
-                          model['plotting']['dist']
-                          )
+
+    model, param_path = parse_parameters(*args)
+
+    u, v, vis, weights = load_data(model)
+
+    if model['modify_data']['norm_by_wle'] or model['modify_data']['cut_data'] or \
+    model['modify_data']['correct_weights']:
+        u, v, vis, weights, wcorr_estimate = alter_data(u, v, vis, weights, model)
+
+    geom = determine_geometry(u, v, vis, weights, model)
+
+    if model['analysis']['bootstrap']:
+        boot_fig, boot_axes = perform_bootstrap(u, v, vis, weights, geom, model)
+        return boot_fig, boot_axes
+
+    else:
+        sol, iteration_diagnostics = perform_fit(u, v, vis, weights, geom, model)
+
+        figs, axes, model = output_results(u, v, vis, weights, sol,
+                                    iteration_diagnostics, model
+                                    )
+
+        logging.info('  Updating {} with final parameters used in'
+                     ' fit'.format(param_path))
+        with open(param_path, 'w') as f:
+            json.dump(model, f, indent=4)
```

Comment:
Why is this at the end?

Reply:
In `fit.py` --> `output_results` --> `if model['plotting']['diag_plot']:` --> `model['plotting']['iter_plot_range'] = iter_plot_range` gets updated based on the final actual iteration of the fit.

This is also a future-proof catchall for any other model parameters that may be updated during a fit.

Reply:
Ok, is it worth writing a temporary one at the start, and overwriting it at the end? So the users can inspect it if the code crashes?

Reply:
I'm not sure that, for this particular case, we want the changes written to file.

Reply:
Good point, it's better not to update the parameter file in this case as the log will say how many iterations were performed anyway. 

I think it's a bit cleaner just to overwrite the same used parameter file instead of creating a temporary one used during the fit, deleting it and dumping to a final one.

Reply:
I'm Ok with that

Reply:
ok cool, the current version does this

Reply:
Is there a temporary json file written and then overwritten?